### ISSUE TITLE
Rewrite/Refactor parts of RESTClient impl.

### ISSUE THREAD

**Issue by syncblaze** [hikari-py/hikari#2022]
### Summary

There is a lot of inconsistency which build up over the last few years in the way the diffrent rest methods are implemented. One big inconsistency for example is handling value of function parameters. 
I will leave a example for this here you can see that the None & undefined check is doubled.
![IMG_0192](https://github.com/user-attachments/assets/f9d8f162-809c-4d64-9473-61313e58d807)

### Requirements
#2021 needs to be merged first before this can makes sense doing

### Why is this needed?

Consistency is nice and doubled checks are bad for developer expierence and prob a small bit performance.

### Ideal implementation

Looking through most of the rest functions and refactor some of them where needed.

### Checklist

- [X] I have searched the issue tracker and have made sure it's not a duplicate. If it is a follow up of another issue, I have specified it.

**Comment by syncblaze** [comment]
this is just a small reminder for me for doing this so dont worry

**Comment by rayakame** [comment]
Fixed in #2021
